src/btree/row_modify.c

```diff
@@ -339,9 +342,23 @@ __wt_update_obsolete_check(
 	 * trim update lists independently of the page state, ensure there
 	 * is a modify structure.
 	 */
-	if (count > 20 && page->modify != NULL)
-		page->modify->obsolete_check_txn =
-		    S2C(session)->txn_global.last_running;
+	if (count > 20 && page->modify != NULL) {
+		page->modify->obsolete_check_txn = txn_global->last_running;
+#ifdef HAVE_TIMESTAMPS
+		if (txn_global->has_pinned_timestamp) {
+			WT_WITH_TIMESTAMP_READLOCK(session, &txn_global->rwlock,
+			    __wt_timestamp_set(
+				&page->modify->obsolete_check_timestamp,
+				&txn_global->pinned_timestamp));
+			page->modify->has_obsolete_check_timestamp = true;
```

Comment:
@sulabhM, can you explain why we need this boolean?

Without it, `page->modify->obsolete_check_timestamp` should be zero, which means it should always pass the `__wt_txn_visible_all` check below.